### ISSUE TITLE
allure-awaitility: the steps in the allure-report break after calling the await() function

### ISSUE THREAD

**Issue by petrschitow** [allure-framework/allure-java#891]
### Brief description of the problem:
I started using the [allure-awaitility](https://github.com/allure-framework/allure-java/tree/master/allure-awaitility) library and saw strange behavior - after the first call to the await() function in the test, either the hierarchy of steps in the report breaks or there are no steps at all after the step that contains await()

### Steps to reproduce:
I created a simple [project](https://github.com/petrschitow/allure-awaitility-issue) to demonstrate the problem (Kotlin + TestNG): https://github.com/petrschitow/allure-awaitility-issue 

To reproduce the error, you need to run the tests and generate an allure-report using the following commands:
`mvn clean test`
`allure serve target/allure-results`

To observe different behavior, run tests with commented/uncommented lines 32-34 in [TestConfiguration.kt](https://github.com/petrschitow/allure-awaitility-issue/blob/main/src/main/kotlin/config/TestConfiguration.kt#L33) file

You can find screenshots with actual/expected behaviour [here](https://github.com/petrschitow/allure-awaitility-issue/tree/main/src/test/resources/screenshots)

P.S. The pet-api is unstable, so you may need to run the test a couple of times.

**Comment by vicusbass** [comment]
This is a major issue, awaitility is a very popular library

**Comment by pme-sap** [comment]
Seems to be caused by awaitility polling with a different thread. Using `await().pollInSameThread()` fixes the issue for me.

**Comment by vicusbass** [comment]
@pme-sap Yeah, seems to do the trick. It does go somehow against what's awaitility supposed to do though, a proper fix would be great

**Comment by ilsilk** [comment]
Hi! Are there any updates on this? It's still reproducible in 2.27.0. Looks quite serious.
